### DISCUSSION TITLE
Attribution REST End point

### DISCUSSION THREAD

**Discussion by mbc32** [GMOD/Apollo#2589] [Ideas]
HI Natham,
It Would be useful with a REST end point for Attributions only. The getHistoryForFeatures does give information about which annotation did the changes but also includes a lot of gene structural information and no information functional annotation.
I was thinking of modifying generateHistory in FeatureEventService (or make a new function) to return attributions for gene edits and functional edits but nothing else so it would be easy to pass. It looks to be simple but wanted to hear what you think.
Thanks for fixing the getHistoryForFeatures endpoint.
Mikkel (mbc32)

**Comment by nathandunn** [comment]
I’m not entirely sure what your returned for would be, but I say its worth a shot. 

I would create two methods though:

1 - one in AnnotationEditorController that mirrors getHistoryForFeatures 
2 - one in FeatureEventService that you’d proposed. 

Obviously in you’d call your new method instead of your old one. 

That being said, it should return functional annotation edits on the genes directly already. 

That being said, it might be missing things somewhere, or maybe you mean something different (also, FYI I just found and fixed this bug https://github.com/GMOD/Apollo/pull/2590 <https://github.com/GMOD/Apollo/pull/2590>). 



Nathan
